### ISSUE TITLE
[20211010] Weekly AI ArXiv 만담

### ISSUE THREAD

**Issue by jungwoo-ha** [jungwoo-ha/WeeklyArxivTalk#26]
- News
  - ICLR Submission: 모두들 수고하셨습니다. - 3700편 (40% 증가)
    - [정리된 구글닥](https://docs.google.com/spreadsheets/d/1lQT-vqvPFF8EESqIabpn9zEOYlw8qDxcecuNjVxwpW0/edit#gid=1543772472) 
  - NeurIPS 2021 Notification 축하드립니다.
  - ICCV 2021: 10.11 ~ 17
  - [Facebook 사태](https://news.naver.com/main/read.naver?mode=LSD&mid=sec&sid1=104&oid=437&aid=0000277718)
  - [한국판 DARPA 첫발, '국방과학' 부처·출연연 다 모였다](http://www.hellodd.com/news/articleView.html?idxno=94530)
- ArXiv
  - [StyleNeRF: A Style-based 3D Aware Generator for High-resolution Image Synthesis](https://openreview.net/forum?id=iUuzzTMUw9K)
    - ICLR 2022 제출 버전 중 단연 화제작
    - NeRF를 이용한 이미지 생성이 대세로 넘어오고 있음
    - 2D 이미지로부터 고해상도 실사 이미지 생성과  multi view의 3d consistency 맞추기 둘다 가능하게 만든 최초의 모델
    - 전체적으로 NeRF + StyleGAN2 + Progressive upsampling 구조
    - Volume rederer 는 low-resolution feature map 생성까지만 씀
    - 그 이후 개선된 upsampler 를 이용해서 해상도를 끌어올리고 regularization term으로 3d consistency 보존
    - Camera pose가 없는 2D 이미지 데이터로 부터 학습하기 위해 camera predictor도 self-supervised 스타일로 학습
![image](https://user-images.githubusercontent.com/11782739/136695378-f7e148b4-3f58-466f-b79e-917a55810f56.png)
![image](https://user-images.githubusercontent.com/11782739/136695448-3a9381af-eeab-485b-be08-22ae5c2d381e.png)
![image](https://user-images.githubusercontent.com/11782739/136695468-c44167a1-79ea-4e14-af2b-3ac2cf89c76c.png)
![image](https://user-images.githubusercontent.com/11782739/136695489-b7fe8b40-247e-4f6b-9669-3d66dd07bfa5.png)
  
  - [Efficient Self-supervised Vision Transformers for Representation Learning](https://openreview.net/forum?id=fVu3o-YUGQK)
    - ICLR 2022 제출버전
    - ViT를 ImageNet-1k 훈련데이터만으로 매우 효율적으로 self-supervised로 학습하는 기법 (DINO나 MoCo-v3는 헤비하다고)
    - Multi-stage architecture with sparse SA --> complexity는 줄여도 region간 fine-grained correspondence 표현 감소
    - 그래서 image region matching하는 pretraining task 추가해서 둘을 결합
    - DINO (self-distill) 스타일의 non-contrastive 
    - ImageNet-1k validset linear probling 기준 81.3% 달성
![image](https://user-images.githubusercontent.com/11782739/136644322-3f2f91ed-10e6-4aa6-86e9-2a76345f55c2.png)
![image](https://user-images.githubusercontent.com/11782739/136644391-98b5b7c4-b2c0-42e5-8085-008c7d9f2878.png)

  - [Scale Efficiently: Insights from Pretraining and Finetuning Transformers](https://openreview.net/forum?id=f2OYVDyfIB)
    - ICLR 2022 제출버전 (Double blind 하지만 구글냄새 솔솔)
    - Pretraining scaling law가 downstream에서 그대로 먹히지 않을 수도 있음.
    - 작은 스케일의 모델에서 찾은 scaling 전략이 큰 모델에서 먹히지 않을 수 있음.
    - 그래서 DeepNarrow라는 전략을 소개함
    - T5 5M ~ 30B까지 활용 LM 뿐 아니라 Vision Transformer에도 실험해봄.
    - 100 개 넘는 T5 체크포인트를 release 하겠다고 (우와 ViT 몇만개 이후 완전 기대)
![image](https://user-images.githubusercontent.com/11782739/136688180-103937be-6d68-45b4-802e-b1bcb9628da8.png)
![image](https://user-images.githubusercontent.com/11782739/136688273-01becfaf-d829-4d69-9656-ad7bbf556b8c.png)

  - [VideoCLIP: Contrastive Pre-training for Zero-shot Video-Text Understanding](https://arxiv.org/abs/2109.14084)
    - Facebook + UW (EMNLP 2021)
    - video 와 자막을 활용하여 pretraining, downstream task에 대해서는 annotation 필요없는 zeroshot.
    - InfoNCE loss + positive pair + hard negative pair
    - Positive: text 고르고 --> 근처 video clip 고르고 --> random 길이, Negative: Retrieval-based sampling
    - Text-to-Video Retrieval, VideoQA, Action segmentation, Action step localization
    - Pretraining HowTo100M 으로부터 추출해서 사용
    - https://github.com/pytorch/fairseq/tree/main/examples/MMPT 
![image](https://user-images.githubusercontent.com/11782739/136689048-102c0472-a277-41cc-a465-e5f14acdbe40.png)

  - [Audio Lottery: Speech Recognition Made Ultra-Lightweight, Noise-Robust, and Transferable](https://openreview.net/forum?id=9Nk6AJkVYB)
    - Lottery ticket for on-device ASR
    - Wining ticket의 경우 충분히 작은 크기로도 noisy에 robust하고 개인화도 되고 
    - CTC, RNN-Trasnducer, Conformer 각각에 대해 실험했을 때 21%, 11%, 8%만 non-zero weight 남겨도 좋은 성능
![image](https://user-images.githubusercontent.com/11782739/136689396-e82b2670-9d78-431f-96bf-c3811ad938d5.png)https://openreview.net/group?id=ICLR.cc/2022/Conference 
![image](https://user-images.githubusercontent.com/11782739/136689450-1281d1ff-84cf-4135-b6a6-52ac9b0998b4.png)

**Comment by veritas9872** [comment]
**PyTorch Universal Docker Template: A Universal PyTorch Source-Build Docker Template**

오늘은 ~~논문을 준비하지 못해서~~ 제 개인 프로젝트를 하나 공유하고자 합니다.
GitHub: https://github.com/veritas9872/PyTorch-Universal-Docker-Template

Deep Learning을 하면서 대다수의 연구자들이 PyTorch를 사용하고 있습니다. 하지만 pip과 conda로 설치하는 PyTorch package는 호환성을 위해서 여러 시스템에서 사용할 수 있는 버전으로 packaging이 되어 있고 사용하는 기계에 최적화되어 있지 않습니다.

물론 PyTorch official Docker image와 NVIDIA NGC PyTorch image 등 여러 source build image가 있지만 실제로 사용해보면 CUDA version 호환성 및 기존 패키지와 사용하고자 하는 패키지가 호환되지 않는 등 여러가지 문제가 발생합니다.

그런 불편함을 해소하고자 제가 이번에 모든 PyTorch version, CUDA version, cuDNN version, Ubuntu version에 대해서 각 사용자가 PyTorch를 원하는 장비에 최적화되도록 source build를 할 수 있는 template을 만들었습니다.

또한, 연구실에서 특히 Docker를 사용하지 않으시는 분들이 많기 때문에 pip install 가능한 wheel을 추출할 수 있도록 만들었습니다.

학교와 산업체 모두에서 매우 유용하게 사용할 수 있다고 생각됩니다. 실제로 benchmarking을 했을 때 4배 가량 빨라졌는데 대학원에서 상당 수의 deep learning project는 pip/conda install 한 PyTorch에서 fp32bit으로 학습을 진행하고 있습니다. 만약 이 template을 사용한다면 연산 속도가 10배 이상 빨라질 수 있을 것으로 기대됩니다.
